AzureMonitorAgent/agent.py

```diff
@@ -0,0 +1,864 @@
+                      'package manager on the VM is currently locked: ' \
+                      'please wait and try again'.format(DPKGLockedErrorCode)
+        elif exit_code is not 0:
+            message = '{0} failed with exit code {1} {2}'.format(operation,
+                                                             exit_code, output)
+
+    except AzureMonitorAgentForLinuxException as e:
+        exit_code = e.error_code
+        message = e.get_error_message(operation)
+    except Exception as e:
+        exit_code = 1
+        message = '{0} failed with error: {1}\n' \
+                  'Stacktrace: {2}'.format(operation, e,
+                                           traceback.format_exc())
+     
+    # Finish up and log messages
+    log_and_exit(operation, exit_code, message)   
+        
+def check_disk_space_availability():
+    """
+    Check if there is the required space on the machine.
+    """
+    try:
+        if get_free_space_mb("/var") < 500 or get_free_space_mb("/etc") < 500 :
+            # 52 is the exit code for missing dependency i.e. disk space
+            # https://github.com/Azure/azure-marketplace/wiki/Extension-Build-Notes-Best-Practices#error-codes-and-messages-output-to-stderr
+            return 52
+        else:
+            return 0
+    except:
+        print('Failed to check disk usage.')
+        return 0
+        
+
+def get_free_space_mb(dirname):
+    """
+    Get the free space in MB in the directory path.
+    """
+    st = os.statvfs(dirname)
+    return st.f_bavail * st.f_frsize / 1024 / 1024
+  
+
+def install():
+    """
+    Ensure that this VM distro and version are supported.
+    Install the Azure Monitor Linux Agent package, using retries.
+    Note: install operation times out from WAAgent at 15 minutes, so do not
+    wait longer.
+    """
+    find_package_manager("Install")
+    exit_if_vm_not_supported('Install')
+
+    # public_settings, protected_settings = get_settings()
+    # if public_settings is None:
+    #     raise ParameterMissingException('Public configuration must be ' \
+    #                                     'provided')
+    package_directory = os.path.join(os.getcwd(), PackagesDirectory)
+    bundle_path = os.path.join(package_directory, BundleFileName)
+    os.chmod(bundle_path, 100)
+    print (PackageManager, "and", BundleFileName)
+    OneAgentInstallCommand = "{0} -i {1}".format(PackageManager, bundle_path)
+    hutil_log_info('Running command "{0}"'.format(OneAgentInstallCommand))
+
+    # Retry, since install can fail due to concurrent package operations
+    exit_code, output = run_command_with_retries_output(OneAgentInstallCommand, retries = 15,
+                                         retry_check = retry_if_dpkg_locked,
+                                         final_check = final_check_if_dpkg_locked)
+
+    try:
+        if os.path.isfile("/etc/default/mdsd"):
+            with open("/etc/default/mdsd", "a") as f:
```

Comment:
can you pls verify that it adds this as expected on a test depl?

Reply:
Yes I can see additions in both rhel and ubuntu test distros